src/scriptworker/cot/verify.py

```diff
@@ -1375,14 +1374,11 @@ async def get_in_tree_template(link):
 
 def _get_action_from_actions_json(all_actions, callback_name):
     for defn in all_actions:
-        if defn.get("kind") == "hook":
-            if defn.get("hookPayload", {}).get("decision", {}).get("action", {}).get("cb_name") == callback_name:
-                return defn
-        elif defn.get("kind") == "task":
-            if defn.get("task", {}).get("$let", {}).get("action", {}).get("cb_name") == callback_name:
-                return defn
-        else:
+        if defn.get("kind") != "hook":
```

Comment:
I wonder if we should move this under the `if` below.

As currently written, it'll bail if we find a non-hook action at all, even if it doesn't match the callback name. This may not be problematic now, but if we want to introduce a new action kind on an experimental basis, it may potentially break all action CoT verification until we update scriptworker. (We had this issue before you touched this code, though, so up to you if you want to make that fix now.)

Reply:
Fixed